decrypt.go

```diff
@@ -30,15 +30,23 @@ func (p7 *PKCS7) Decrypt(cert *x509.Certificate, pkey crypto.PrivateKey) ([]byte
 	if recipient.EncryptedKey == nil {
 		return nil, errors.New("pkcs7: no enveloped recipient for provided certificate")
 	}
-	switch pkey := pkey.(type) {
-	case *rsa.PrivateKey:
+	if pkey, ok := pkey.(*rsa.PrivateKey); ok {
 		var contentKey []byte
 		contentKey, err := rsa.DecryptPKCS1v15(rand.Reader, pkey, recipient.EncryptedKey)
 		if err != nil {
 			return nil, err
 		}
 		return data.EncryptedContentInfo.decrypt(contentKey)
 	}
+
+	if pkey, ok := pkey.(crypto.Decrypter); ok {
+		var contentKey []byte
+		contentKey, err := pkey.Decrypt(rand.Reader, recipient.EncryptedKey, nil)
+		if err != nil {
+			return nil, err
+		}
+		return data.EncryptedContentInfo.decrypt(contentKey)
+	}
```

Comment:
You should be able to do this with a switch too.

Reply:
Thanks! I'm fairly new to Golang and I wasn't sure if a case on a switch var.(type) only supported exact matches or any compatible type. I assumed the former, but if the latter is true, that makes this a lot easier. I'll give it a shot in the morning.